test/k8sT/Services.go

```diff
@@ -317,43 +359,62 @@ var _ = Describe("K8sServicesTest", func() {
-			httpSVCURL := fmt.Sprintf("http://%s/", clusterIP)
-			tftpSVCURL := fmt.Sprintf("tftp://%s/hello", clusterIP)
-
-			status := kubectl.ExecInHostNetNS(context.TODO(), k8s1NodeName,
-				helpers.CurlFail(httpSVCURL))
-			status.ExpectSuccess("cannot curl to service IP from host")
-
-			status = kubectl.ExecInHostNetNS(context.TODO(), k8s1NodeName,
-				helpers.CurlFail(tftpSVCURL))
-			status.ExpectSuccess("cannot curl to service IP from host")
 			ciliumPods, err := kubectl.GetCiliumPods()
 			Expect(err).To(BeNil(), "Cannot get cilium pods")
-			for _, pod := range ciliumPods {
-				service := kubectl.CiliumExecMustSucceed(context.TODO(), pod, "cilium service list", "Cannot retrieve services on cilium Pod")
-				service.ExpectContains(clusterIP, "ClusterIP is not present in the cilium service list")
+
+			for _, svcName := range serviceNames {
+				clusterIP, _, err := kubectl.GetServiceHostPort(helpers.DefaultNamespace, svcName)
+				Expect(err).Should(BeNil(), "Cannot get service %s", svcName)
+				Expect(govalidator.IsIP(clusterIP)).Should(BeTrue(), "ClusterIP is not an IP")
+
+				By("testing connectivity via cluster IP %s", clusterIP)
+
+				httpSVCURL := fmt.Sprintf("http://%s/", net.JoinHostPort(clusterIP, "80"))
+				tftpSVCURL := fmt.Sprintf("tftp://%s/hello", net.JoinHostPort(clusterIP, "69"))
+
+				status := kubectl.ExecInHostNetNS(context.TODO(), k8s1NodeName,
+					helpers.CurlFail(httpSVCURL))
+				Expect(status).Should(helpers.CMDSuccess(), "cannot curl to service IP from host: %s", status.CombineOutput())
+
+				status = kubectl.ExecInHostNetNS(context.TODO(), k8s1NodeName,
+					helpers.CurlFail(tftpSVCURL))
+				Expect(status).Should(helpers.CMDSuccess(), "cannot curl to service IP from host: %s", status.CombineOutput())
+
+				for _, pod := range ciliumPods {
+					service := kubectl.CiliumExecMustSucceed(context.TODO(), pod, "cilium service list", "Cannot retrieve services on cilium Pod")
+					service.ExpectContains(clusterIP, "ClusterIP is not present in the cilium service list")
+				}
+				// Send requests from "app2" pod which runs on the same node as
+				// "app1" pods
+				testCurlFromPods("id=app2", httpSVCURL, 10, 0)
+				testCurlFromPods("id=app2", tftpSVCURL, 10, 0)
 			}
-			// Send requests from "app2" pod which runs on the same node as
-			// "app1" pods
-			testCurlFromPods("id=app2", httpSVCURL, 10, 0)
-			testCurlFromPods("id=app2", tftpSVCURL, 10, 0)
 		})
 
 		It("Checks service accessing itself (hairpin flow)", func() {
-			clusterIP, _, err := kubectl.GetServiceHostPort(helpers.DefaultNamespace, echoServiceName)
-			Expect(err).Should(BeNil(), "Cannot get service %q ClusterIP", echoServiceName)
-			Expect(govalidator.IsIP(clusterIP)).Should(BeTrue(), "ClusterIP is not an IP")
+			serviceNames := []string{echoServiceName}
+			// Hairpin flow mode is currently not supported for IPv6.
+			// TODO: Uncomment after https://github.com/cilium/cilium/pull/14138 is merged
+			// if dualStackSupportEnabled {
```

Comment:
This looks odd, we can't "know" the order the variable will be initialized so it would be better executing `helpers.DualStackSupported()` directly here.